### PR TITLE
Reset accumulated text

### PR DESCRIPTION
**Pull request by hugobho**
As a developer dealing with streams of content
  I want to be able to clear all previously accumulated text
  So that I may search through only new and relevant data

### CODE REVIEW

File: features/reset.feature [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Feature: Resetting text accumulator
+
+  As a developer dealing with streams of content
+  I want to be able to clear all previously accumulated text
+  So that I may search through only new and relevant data
```
Comment (kevgo) [inline features/reset.feature]: Please add a period at the end.

File: features/steps/steps.ls [hunk 2]
```diff
@@ -11,10 +11,22 @@ module.exports = ->
     @stream = new ReadableStream ''
     @accumulator = new TextStreamAccumulator @stream
 
+  @Given /^an accumulated text stream$/, (done) ->
+    @stream = new ReadableStream ''
```
Comment (kevgo) [inline features/steps/steps.ls]: This variable doesn't have to be shared between steps. You can remove the `@`

File: features/steps/steps.ls [hunk 3]
```diff
@@ -11,10 +11,22 @@ module.exports = ->
...
+    @accumulator = new TextStreamAccumulator @stream
+    @stream.append \hello
+    @stream.append \world
+    done!
```
Comment (kevgo) [inline features/steps/steps.ls]: This step doesn't have to wait for asynchronous callbacks before it is done. So please remove the `done` method here and in the parameter list.

File: features/steps/steps.ls [hunk 4]
```diff
@@ -11,10 +11,22 @@ module.exports = ->
...
 
   @When /^its incoming stream emits "([^"]*)"$/, (emitted-text) ->
     @stream.append emitted-text
 
+  @When /^I call the reset method$/, (done) ->
+    @accumulator.reset!
+    done!
```
Comment (kevgo) [inline features/steps/steps.ls]: Remove `done`

File: features/steps/steps.ls [hunk 5]
```diff
@@ -11,10 +11,22 @@ module.exports = ->
...
 
   @Then /^its accumulated text is "([^"]*)"$/, (expected-text) ->
     expect(@accumulator.toString!).to.equal expected-text
+
+  @Then /^its accumulated text is empty$/, ->
+    expect(@accumulator.toString!).to.equal ""
```
Comment (kevgo) [inline features/steps/steps.ls]: ...to.be.empty

File: src/text-stream-accumulator.ls [hunk 6]
```diff
@@ -13,6 +13,9 @@ class TextStreamAccumulator
   to-string: ->
     @_accumulated-text
 
+  reset: ->
+    @_accumulated-text := ''
```
Comment (kevgo) [inline src/text-stream-accumulator.ls]: `:=` is for assigning to a variable from a higher scope. You aren't doing this here, but assign to a member variable of the class. Remove the `:`

**Comment by kevgo** [comment]
Cool, thanks for doing this!

File: features/reset.feature [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  I want to be able to clear all previously accumulated text
+  So that I may search through only new and relevant data.
+
+  Rules:
+  - call ".reset()" on the accumulator to clear it of all previously accumulated text
+
+  Scenario: clear accumulated text stream
+    Given an accumulated text stream
+    When I call the reset method
+    Then its accumulated text is empty
```
Comment (kevgo) [inline features/reset.feature]: Add newline at the end here

File: features/reset.feature [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+  As a developer dealing with streams of content
+  I want to be able to clear all previously accumulated text
+  So that I may search through only new and relevant data.
+
+  Rules:
+  - call ".reset()" on the accumulator to clear it of all previously accumulated text
+
+  Scenario: clear accumulated text stream
+    Given an accumulated text stream
+    When I call the reset method
```
Comment (kevgo) [inline features/reset.feature]: Surround reset in double quotes.

File: features/steps/steps.ls [hunk 9]
```diff
@@ -11,10 +11,20 @@ module.exports = ->
     @stream = new ReadableStream ''
     @accumulator = new TextStreamAccumulator @stream
 
+  @Given /^an accumulated text stream$/, ->
```
Comment (kevgo) [inline features/steps/steps.ls]: Given a text stream accumulator with some content

File: features/steps/steps.ls [hunk 10]
```diff
@@ -11,10 +11,20 @@ module.exports = ->
...
+    stream = new ReadableStream ''
+    @accumulator = new TextStreamAccumulator stream
+    stream.append \hello
+    stream.append \world
 
   @When /^its incoming stream emits "([^"]*)"$/, (emitted-text) ->
     @stream.append emitted-text
 
+  @When /^I call the reset method$/, ->
```
Comment (kevgo) [inline features/steps/steps.ls]: When using double quotes, you get the method name as a string. Use meta-programming to call it here:

```ls
@accumulator[method-name]!

**Comment by hugobho** [comment]
Hey @kevgo could you take a look at the changes I made?

File: features/reset.feature [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+  As a developer dealing with streams of content
+  I want to be able to clear all previously accumulated text
+  So that I may search through only new and relevant data.
+
+  Rules:
+  - call ".reset()" on the accumulator to clear it of all previously accumulated text
+
+  Scenario: clear accumulated text stream
+    Given a text stream accumulator with some content
+    When I calling its "reset" method
```
Comment (kevgo) [inline features/reset.feature]: Remove the "I"

**Comment by kevgo** [comment]
Looks good after the last change is made! 🚢

**Pull request closed without merging**